src/core/FilePath.cpp

```diff
@@ -211,6 +211,9 @@ FilePath::FilePath()
     else if (testSetDir(appDirPath + "/share")) {
     }
 #endif
+    // Last ditch test when running in the build directory (mainly for travis tests)
+    else if (testSetDir(QString(KEEPASSX_SOURCE_DIR) + "/share")) {
```

Comment:
Well, in Debug mode, this is always evaluated as the first option. In Release mode, this path only exists in development environments, though. Checking other paths first is not a bad idea since it saves quite some file checks.